include/alpaka/kernel/Traits.hpp

```diff
@@ -222,6 +222,9 @@ namespace alpaka
         // check for void return type
         detail::CheckFnReturnType<TAcc>{}(kernelFnObj, args...);
 
+        static_assert(
+            std::conjunction_v<std::is_trivially_copyable<std::decay_t<TArgs>>...>,
+            "Kernel arguments must be trivially copyable!");
```

Comment:
```suggestion
        static_assert(
            (std::is_trivially_copyable_v<std::decay_t<TArgs>> && ... && true),
            "Kernel arguments must be trivially copyable!");
```
I prefer a fold, but up to you!

Reply:
From what I understand a fold instantiates all parameters while `std::conjunction` will look at them individually (and fail at the first `false`).

Reply:
By the way, the final `&& true` is not necessary.

Reply:
Maybe, but it should not matter. Anyway, do what you prefer! :)

Reply:
> By the way, the final && true is not necessary.

That would not work for kernels without arguments, which are probably rare or even nonsensical. But why prevent it?

Reply:
Mhm, I think it should work also with an empty argument list: https://godbolt.org/z/TP99M1nfh . 

[Cppreference](https://en.cppreference.com/w/cpp/language/fold#Explanation) says
> When a unary fold is used with a pack expansion of length zero, only the following operators are allowed:
> 1) Logical AND (`&&`). The value for the empty pack is `true`

Reply:
> From what I understand a fold instantiates all parameters while `std::conjunction` will look at them individually (and fail at the first `false`).

I think the fold expression itself does follow short-circuit evaluation: https://godbolt.org/z/e3jbbh4d3 .
The values after the first `false` are not evaluated.

Reply:
I was following the info from this link: https://stackoverflow.com/questions/55161235/is-there-a-reason-to-use-stdconjunction-stddisjunction-instead-of-a-fold-exp#comment97065030_55161485

Reply:
But anyway, I don't think this will hurt our compilation performance much. And having a language feature instead of a library feature is also nice, so I'll change to a fold expression.

Reply:
Ah, I see. Yes, you are right, the arguments are instantiated, even if they are not evaluated: https://godbolt.org/z/osjbc98an .